### PR TITLE
Drop py33

### PR DESCRIPTION
**Pull request by alex**
…upport it

Refs #676

### CODE REVIEW

**Comment by alex** [comment]
@hynek this PR is _not_ fully dropping support, it's just dropping support when using versions of cryptography that don't support it. If you want me to fully drop support, let me know.

**Comment by hynek** [comment]
So on one hand: it’s rare that we manage to release a new pyOpenSSL without requiring newer bindings eventually.

OTOH, nobody uses Python 3.3 but Twisted seems to support it anyways.  So let’s ask @glyph if our dropping would be a welcome excuse to drop it too or whether it’ll make him sad.

**Comment by alex** [comment]
Dropping it from cryptography was based on a review of the data, and its extremely low usage. IIRC, pyopenssl it is like .1% of downloads.

**Comment by glyph** [comment]
IIRC @hawkowl has proposed dropping 3.3 and it's already on the slate of things to do because our metrics are similarly tiny.  3.4 is the first version of python 3, I'm pretty sure the memory hole will swallow up <= 3.3 in due time.

**Comment by hynek** [comment]
If you please @alex... 🔥

**Comment by alex** [comment]
🔥 🔥 🔥  MY PLEASURE 🔥 🔥 🔥

**Comment by alex** [comment]
@hynek done.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -12,7 +12,8 @@ The third digit is only for regressions.
 Backward-incompatible changes:
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-*none*
+- Dropped support for Python 3.3.
+  `#677 <https://github.com/pyca/pyopenssl/pull/677`_
```
Comment (hynek) [inline CHANGELOG.rst]: This lacks the closing `>` which is breaking CI.

**Comment by hynek** [comment]
bye bye 🔥

**Pull request merged**